src/main/scala/models/DataCell.scala

```diff
@@ -20,6 +20,16 @@ case class DataCell(name: String, Type: ArcaneType, value: Any)
  */
 object DataCell:
   def apply(name: String, Type: ArcaneType, value: Any): DataCell = new DataCell(name, Type, value)
-  
+
+  /**
+   * Extension method to get the schema of a DataRow.
+   */
+  extension (row: DataRow) def schema: ArcaneSchema =
+    row.foldLeft(ArcaneSchema.empty()) {
+      case (schema, cell) if cell.name == MergeKeyField.name => schema ++ Seq(MergeKeyField)
+      case (schema, cell) if cell.name == DatePartitionField.name => schema ++ Seq(DatePartitionField)
```

Comment:
`DatePartitionField` can be removed from everywhere btw - it is not used and won't be

Reply:
Will do it in next PRs